bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/ResponseMappings.kt

```diff
@@ -37,4 +123,42 @@ fun ChangelogEntry.toGateDto(): ChangelogResponse {
     )
 }
 
+fun LegalEntityGateInputRequest.toLegalEntity(): LegalEntity {
+
+    val addressInputRequest =AddressGateInputRequest(
+        address= legalEntity.legalAddress,
+        externalId= externalId+"_legalAddress",
```

Comment:
Best would be to create a simple service function that returns a legal address external id for a given legal entity external id. You can use this function here then to assign a legal address external id